New package: netatalk-3.1.15

bsdelf opened the pull request:
<!-- Uncomment relevant sections and delete options which are not applicable -->

#### Testing the changes
- I tested the changes in this PR: **YES**


#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

<!-- Note: If the build is likely to take more than 2 hours, please add ci skip tag as described in
https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration
and test at least one native build and, if supported, at least one cross build.
Ignore this section if this PR is not skipping CI.
-->

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)

Duncaen commented on srcpkgs/netatalk/template:
```diff
@@ -0,0 +1,25 @@
+pkgname=netatalk
+version=3.1.15
+revision=1
+archs='x86_64'
```
This should probably not be restricted to only one architecture.

bsdelf replied:
I've tried `-a aarch64` and `-a armv6l-musl`, both of them failed at the configure stage

```
checking for SSL... no
checking for Berkeley DB headers in /usr/local/include/... no
checking for Berkeley DB headers in /usr/local/include/db6.1... no
checking for Berkeley DB headers in /usr/local/include/db6... no
checking for Berkeley DB headers in /usr/local/include/db5... no
checking for Berkeley DB headers in /usr/local/include/db5.3... no
checking for Berkeley DB headers in /usr/local/include/db5.2... no
checking for Berkeley DB headers in /usr/local/include/db5.1... no
checking for Berkeley DB headers in /usr/local/include/db51... no
checking for Berkeley DB headers in /usr/local/include/db5.0... no
checking for Berkeley DB headers in /usr/local/include/db50... no
checking for Berkeley DB headers in /usr/local/include/db4.8... no
checking for Berkeley DB headers in /usr/local/include/db48... no
checking for Berkeley DB headers in /usr/local/include/db4.7... no
checking for Berkeley DB headers in /usr/local/include/db47... no
checking for Berkeley DB headers in /usr/local/include/db4.6... no
checking for Berkeley DB headers in /usr/local/include/db46... no
checking for Berkeley DB headers in /usr/local/include/db4... no
checking for Berkeley DB headers in /usr/include/... no
checking for Berkeley DB headers in /usr/include/db6.1... no
checking for Berkeley DB headers in /usr/include/db6... no
checking for Berkeley DB headers in /usr/include/db5... no
checking for Berkeley DB headers in /usr/include/db5.3... no
checking for Berkeley DB headers in /usr/include/db5.2... no
checking for Berkeley DB headers in /usr/include/db5.1... no
checking for Berkeley DB headers in /usr/include/db51... no
checking for Berkeley DB headers in /usr/include/db5.0... no
checking for Berkeley DB headers in /usr/include/db50... no
checking for Berkeley DB headers in /usr/include/db4.8... no
checking for Berkeley DB headers in /usr/include/db48... no
checking for Berkeley DB headers in /usr/include/db4.7... no
checking for Berkeley DB headers in /usr/include/db47... no
checking for Berkeley DB headers in /usr/include/db4.6... no
checking for Berkeley DB headers in /usr/include/db46... no
checking for Berkeley DB headers in /usr/include/db4... no
configure: error: Berkeley DB library required but not found!
```

Duncaen commented:
```diff
diff --git a/srcpkgs/netatalk/template b/srcpkgs/netatalk/template
index b05617ace0..82d3688a5c 100644
--- a/srcpkgs/netatalk/template
+++ b/srcpkgs/netatalk/template
@@ -2,24 +2,28 @@
 pkgname=netatalk
 version=3.1.15
 revision=1
-archs='x86_64'
 build_style=gnu-configure
-conf_files="
-       /etc/dbus-1/system.d/netatalk-dbus.conf
-       /etc/extmap.conf
-       /etc/afp.conf
-       /etc/dbus-session.conf"
-hostmakedepends="autoconf automake pkg-config tar"
+hostmakedepends="automake pkg-config"
 makedepends="libgcrypt-devel db-devel libevent-devel avahi-libs-devel openssl-devel dbus-glib-devel"
 depends="python3"
 short_desc="Open Source fileserver that implements the Apple Filing Protocol (AFP)"
 maintainer="Yanhui Shen <shen.elf@gmail.com>"
 license="GPL-2.0-only"
 homepage="https://github.com/Netatalk/netatalk"
-distfiles="${homepage}/releases/download/${pkgname}-${version//./-}/netatalk-${version}.tar.bz2"
+distfiles="https://github.com/Netatalk/netatalk/releases/download/netatalk-${version//./-}/netatalk-${version}.tar.bz2"
 checksum=d8d4a5cc0fd86b0d90e017d64c1f4623e8cdbfbda570fc423ade1151a93d19f5
 python_version=3

+conf_files="
+ /etc/dbus-1/system.d/netatalk-dbus.conf
+ /etc/extmap.conf
+ /etc/afp.conf
+ /etc/dbus-session.conf"
+
+if [ "$CROSS_BUILD" ]; then
+       configure_args+=" --with-bdb=${XBPS_CROSS_BASE}"
+fi
+
 post_install() {
        vsv netatalk
 }

```

Chocimier commented on srcpkgs/netatalk/template:
```diff
@@ -0,0 +1,25 @@
+	/etc/dbus-1/system.d/netatalk-dbus.conf
+	/etc/extmap.conf
+	/etc/afp.conf
+	/etc/dbus-session.conf"
```
Taking over such generic file by random package look like bad idea, especially as current xbps will overwrite if it existed. Maybe install it in /usr/share/examples?

bsdelf replied:
The "dbus-session.conf" can be relocated, other files should be fine. FreeBSD add suffix ".sample" to avoid the file conflict:

```
# pkg info -lx netatalk | grep -e etc -e share
        /usr/local/etc/afp.conf.sample
        /usr/local/etc/extmap.conf.sample
        /usr/local/etc/pam.d/netatalk.sample
        /usr/local/etc/rc.d/netatalk
        /usr/local/share/aclocal/netatalk.m4
        /usr/local/share/licenses/netatalk3-3.1.15,1/GPLv2
        /usr/local/share/licenses/netatalk3-3.1.15,1/LICENSE
        /usr/local/share/licenses/netatalk3-3.1.15,1/catalog.mk
```

> especially as current xbps will overwrite if it existed

I edited the afp.conf and then reinstalled the package, the afp.conf is not overwrote.

Chocimier replied:
> afp.conf is not overwrote

OK, sorry, new installs are fine and overwrite happens in slightly different conditions.

bsdelf commented:
Should I fix the lint?  Why is space not preferred by the way?

```
Error: please indent with tabs
```


> ```diff
> -conf_files="
> -       /etc/dbus-1/system.d/netatalk-dbus.conf
> -       /etc/extmap.conf
> -       /etc/afp.conf
> -       /etc/dbus-session.conf"
> 
> +conf_files="
> + /etc/dbus-1/system.d/netatalk-dbus.conf
> + /etc/extmap.conf
> + /etc/afp.conf
> + /etc/dbus-session.conf"
> ```

Chocimier commented:
Preferred style is single space for variable continuation and tab in code blocks and functions. conf_files is fine, lint points at if block. Please adjust.

bsdelf commented:
Updated and checked with xlint. Now it should be good.

Duncaen reviewed (changes requested):
The header files, static libraries, m4 files and netatalk-config should be split into a `-devel` subpackage.

bsdelf commented:
Updated. The devel files are splitted, except "usr/lib/netatalk/*.so", they are dynamically loaded at runtime based on the configuration. Also I fixed the start script.

bsdelf commented:
Oh no, the current start script still has a issue. The afpd process no longer starts with following code: 

```
exec chpst -b netatalk pause
```

The tricky thing is `netatalk` does not support foreground running mode, is there a "standard way" to handle it in void runit?

bsdelf commented:
> The tricky thing is `netatalk` does not support foreground running mode, is there a "standard way" to handle it in void runit?

I fixed this issue by adding `-d` option to `netatalk` command, it's not mentioned in manual or help message, I found it in the source code, it prevents daemonize, which is required by runit.

Now the `sv start` and `sv stop` command is working, and the network location showing & disappering can be observed in macOS immediately.
